### PR TITLE
Include the correct use of io.socket.request

### PR DESCRIPTION
**Pull request by JemiloII**
The old example demonstrates how to use io.socket._request

### CODE REVIEW

File: reference/websockets/sails.io.js/socket.request.md [hunk 1]
```diff
@@ -10,7 +10,30 @@ This function is very similar to `io.socket.get()`, `io.socket.post()`, etc. exc
 ### Usage
 
 ```js
-io.socket.request(options, function (data, jwr)){
+io.socket.request(url, reqData, function (resData, jwr)){
```
Comment (jub3i) [inline reference/websockets/sails.io.js/socket.request.md]: 1. Syntax error, one too many closing parentheses.
2. `request()` does not have that function signature: https://github.com/balderdashy/sails-generate-sails.io.js/blob/master/templates/sails.io.js#L847

File: reference/websockets/sails.io.js/socket.request.md [hunk 2]
```diff
@@ -10,7 +10,30 @@ This function is very similar to `io.socket.get()`, `io.socket.post()`, etc. exc
...
+  // ...
+  // jwr.headers
+  // jwr.statusCode
+  // jwr.body === data
+  // ...
+}, 'post');
+```
+
+```js
+io.socket._request(options, function (data, jwr)){
```
Comment (jub3i) [inline reference/websockets/sails.io.js/socket.request.md]: `_request()` is deprecated, see: https://github.com/balderdashy/sails-generate-sails.io.js/blob/master/templates/sails.io.js#L924

File: reference/websockets/sails.io.js/socket.request.md [hunk 3]
```diff
@@ -23,10 +46,10 @@ io.socket.request(options, function (data, jwr)){
 ### Example
 
 ```javascript
-io.socket.request({
+io.socket._request({
```
Comment (jub3i) [inline reference/websockets/sails.io.js/socket.request.md]: `_request()` is deprecated, see: https://github.com/balderdashy/sails-generate-sails.io.js/blob/master/templates/sails.io.js#L924

File: reference/websockets/sails.io.js/socket.request.md [hunk 4]
```diff
@@ -23,10 +46,10 @@ io.socket.request(options, function (data, jwr)){
...
   method: 'get',
   url: '/user/3/friends',
-  params: {},
+  data: {},
```
Comment (jub3i) [inline reference/websockets/sails.io.js/socket.request.md]: first argument options needs property `params` not `data` https://github.com/balderdashy/sails-generate-sails.io.js/blob/master/templates/sails.io.js#L871

**Comment by jub3i** [comment]
There are a large number or errors here and even if they where cleaned up I don't see what additional value would be contributed. I'm closing this for now, please comment if you think I've missed something.

**Comment by JemiloII** [comment]
I've been using the _request, i've been enjoying it's ease of use

**Comment by jub3i** [comment]
I normally just go with `io.socket.post()` or whatever method I need to use, are you doing lots of custom requests? I would be careful to use it too much as if you upgrade to Sails v0.11 it won't work any more...

**Comment by JemiloII** [comment]
yeah, i like to include headers, body, params, query, and method all in one go. The other methods aren't nice to use when you need headers. Also, http allows get request to send data, I take advantage of this time to time.

**Pull request closed without merging**